src/lib.rs

```diff
@@ -136,13 +136,17 @@ pub async fn start_lemmy_server() -> Result<(), LemmyError> {
 
   // Create Http server with websocket support
   HttpServer::new(move || {
-    let cors_config = if cfg!(debug_assertions) {
-      Cors::permissive()
-    } else {
-      let cors_origin = std::env::var("LEMMY_CORS_ORIGIN").unwrap_or("http://localhost".into());
-      Cors::default()
+    let cors_origin = std::env::var("LEMMY_CORS_ORIGIN");
+    let cors_config = match cors_origin {
+      Ok(cors_origin) => Cors::default()
         .allowed_origin(&cors_origin)
-        .allowed_origin(&settings.get_protocol_and_hostname())
+        .allowed_origin(&settings.get_protocol_and_hostname()),
+      Err(_) => Cors::default()
+        .allow_any_origin()
+        .allow_any_method()
+        .allow_any_header()
+        .expose_any_header()
+        .max_age(3600),
```

Comment:
Why are you adding a max age here?

Reply:
A `max-age` allows the browser to cache/skip checking for CORS again in that period of time.

Without it, the browser will have to do one request before every `POST` request every single time.

Reply:
Okay, as long as it only affects CORS, and not anything else.